serial_linux.go

```diff
@@ -62,10 +62,26 @@ func openPort(name string, baud int) (rwc io.ReadWriteCloser, err error) {
 	}()
 
 	fd := f.Fd()
+	// set blocking / non-blocking read
+	var minBytesToRead uint8 = 1
+	if readTimeout > 0 {
+		// EOF on zero read
+		minBytesToRead = 0
+		// capping the timeout
+		if readTimeout < 100 {
+			// min possible timeout
+			readTimeout = 100
+		} else if readTimeout > 25500 {
+			// max possible timeout
+			readTimeout = 25500
+		}
+		// convert milliseconds to deciseconds as expected by VTIME
+		readTimeout = readTimeout / 100
```

Comment:
I think the intention and reason for the clamping would be clearer to do the /100 first and then clamp the values to 1 and math.MaxUint8.